heal/vlmd/extract/extract.py

```diff
@@ -62,26 +70,87 @@ def vlmd_extract(
         logger.error(message)
         raise ExtractionError(message)
 
+    logger.debug(f"File type is set to '{file_type}'")
     # validate
     try:
+        # csv files are converted as part of validate
         converted_dictionary = vlmd_validate(
-            input_file, output_type=output_type, return_converted_output=True
+            input_file,
+            file_type=file_type,
+            output_type=output_type,
+            return_converted_output=True,
         )
+        logger.debug("Converted dict")
     except Exception as e:
         logger.error(f"Error in validating and extracting dictionary from {input_file}")
         logger.error(e)
         raise ExtractionError(str(e))
 
+    # input json file require explicit conversion and post validation steps
+    if file_type == "json":
+        file_convert_function = file_type_to_fxn_map.get(file_type)
+        data_dictionary_props = {}
+        try:
+            logger.debug("Ready to convert json input to VLMD")
+            data_dictionaries = convert_to_vlmd(
+                input_filepath=input_file,
+                input_type=file_convert_function,
+                data_dictionary_props=data_dictionary_props,
+            )
+            if output_type == "json":
+                converted_dictionary = data_dictionaries["template_json"]
+                logger.debug(
+                    f"Ready to validate converted dict with output type '{output_type}'"
+                )
+                logger.debug(
+                    f"Data is pathlike {isinstance(converted_dictionary, (str, os.PathLike))}"
+                )
+                is_valid = vlmd_validate(
+                    converted_dictionary,
+                    file_type=file_type,
+                    output_type=output_type,
+                    return_converted_output=False,
+                )
+                logger.debug(f"Converted dictionary is valid: {is_valid}")
+            else:
+                converted_dictionary = data_dictionaries["template_csv"]["fields"]
+        except Exception as e:
+            logger.error(f"Error in extracting JSON dictionary from {input_file}")
+            logger.error(e)
+            raise ExtractionError(str(e))
+
+    instrument_title = None
+    if output_type == "json":
+        standards_mappings = converted_dictionary.get("standardsMappings")
+        if standards_mappings and len(standards_mappings) >= 1:
+            instrument_title = standards_mappings[0].get("instrument").get("title")
```

Comment:
need some error handling here in case like `standards_mappings` is None or `instrument` is None
better surrond it with try-catch

Reply:
Yes, good idea. This block is now wrapped in a `try` statement. It will log a warning if the `standardsMappings` is malformed (missing `instrument` or `title`). The `title` logic has been moved into a method at the top of the module.